Automattic/wp-calypso
# Modularise stats state

This PR is one of many working on modularising state in Calypso. This one handles stats.

For mode details on state modularisation, see #39261 and p4TIVU-9lM-p2.

#### Changes proposed in this Pull Request

* Modularise stats state
* Move a few selectors under `state/stats`

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read`.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyze the tree and note that there's no `stats` key, even if you expand the list of keys. This indicates that the Stats state hasn't been loaded yet.
* Click `My Sites` on the masterbar, and then navigate into `Stats`.
* Verify that a `stats` key is added with the Stats state. This indicates that the Stats state has now been loaded.
* Verify that Stats work normally. This indicates that I didn't mess up.
